projects/plutus/src/ltc_afe.c

```diff
@@ -0,0 +1,286 @@
+    config_packet.devices[curr_device].reg.discharge_c4 = s_discharging_cells[cell_index + 3];
+    config_packet.devices[curr_device].reg.discharge_c5 = s_discharging_cells[cell_index + 4];
+    config_packet.devices[curr_device].reg.discharge_c6 = s_discharging_cells[cell_index + 5];
+    config_packet.devices[curr_device].reg.discharge_c7 = s_discharging_cells[cell_index + 6];
+    config_packet.devices[curr_device].reg.discharge_c8 = s_discharging_cells[cell_index + 7];
+    config_packet.devices[curr_device].reg.discharge_c9 = s_discharging_cells[cell_index + 8];
+    config_packet.devices[curr_device].reg.discharge_c10 = s_discharging_cells[cell_index + 9];
+    config_packet.devices[curr_device].reg.discharge_c11 = s_discharging_cells[cell_index + 10];
+    config_packet.devices[curr_device].reg.discharge_c12 = s_discharging_cells[cell_index + 11];
+
+    config_packet.devices[curr_device].reg.discharge_timeout = LTC_AFE_DISCHARGE_TIMEOUT_1_MIN;
+
+    config_packet.devices[curr_device].reg.adcopt = ((afe->adc_mode + 1) > 3);
+    config_packet.devices[curr_device].reg.swtrd = true;
+
+    config_packet.devices[curr_device].reg.undervoltage = undervoltage;
+    config_packet.devices[curr_device].reg.overvoltage = overvoltage;
+
+    // GPIO5, ..., GPIO2 are used to MUX data
+    config_packet.devices[curr_device].reg.gpio1 = ((enable >> 3) >> 0) & 1;
+    config_packet.devices[curr_device].reg.gpio2 = ((enable >> 3) >> 1) & 1;
+    config_packet.devices[curr_device].reg.gpio3 = ((enable >> 3) >> 2) & 1;
+    config_packet.devices[curr_device].reg.gpio4 = ((enable >> 3) >> 3) & 1;
+    config_packet.devices[curr_device].reg.gpio5 = ((enable >> 3) >> 4) & 1;
+
+    uint16_t cfgr_pec = crc15_calculate((uint8_t *)&config_packet.devices[curr_device].reg, 6);
+    config_packet.devices[curr_device].pec = SWAP_UINT16(cfgr_pec);
+  }
+
+  prv_wakeup_idle(afe);
+  return spi_exchange(afe->spi_port, (uint8_t *)&config_packet,
+                      sizeof(LTCAFEWriteConfigPacket), NULL, 0);
+}
+
+static StatusCode prv_read_config(const LTCAFESettings *afe,
+                                  LTCAFEConfigRegisterData *configuration_registers) {
+  StatusCode status = STATUS_CODE_OK;
+
+  prv_wakeup_idle(afe);
+
+  LTCAFEWriteDeviceConfigPacket received_data[PLUTUS_AFE_DEVICES_IN_CHAIN] = { 0 };
+
+  size_t len = sizeof(received_data);
+  prv_read_register(afe, LTC_AFE_REGISTER_CONFIG, (uint8_t *)received_data, len);
+
+  for (uint8_t device = 0; device < PLUTUS_AFE_DEVICES_IN_CHAIN; ++device) {
+    configuration_registers[device] = received_data[device].reg;
+
+    uint16_t received_pec = SWAP_UINT16(received_data[device].pec);
+    uint16_t calculated_pec = crc15_calculate((uint8_t *)&received_data[device].reg,
+                                              sizeof(LTCAFEConfigRegisterData));
+    if (calculated_pec != received_pec) {
+      status = status_code(STATUS_CODE_UNKNOWN);
```

Comment:
I was using ``STATUS_CODE_UNKNOWN`` in the sense that it is _unknown_ to the caller whether the data received is valid. I figured that it might be useful for the caller to at least have an idea as to what the invalid CRC-failing data looked like, but sure, exiting early :100: :ok_hand:.